Transaction too big errors

Hi, I started running v0.12.0 and instantly noticed the following errors which get repeated indefinitely for the `/allocs` endpoint for a thaos-query scrape target.

Note there are no similar issues for the other scrape targets. 

```bash
level=error name=parca ts=2022-07-20T16:35:00.852555555Z caller=scrape.go:500 scrape_pool=rhobs target=http://10.128.2.183:9090/debug/pprof/allocs msg="WriteRaw failed for scraped profile" err="rpc error: code = Internal desc = failed to ingest profile: normalize stacktraces: Txn is too big to fit into one request"

level=error name=parca ts=2022-07-20T16:35:36.216191691Z caller=scrape.go:500 scrape_pool=rhobs target=http://10.131.4.81:9090/debug/pprof/allocs msg="WriteRaw failed for scraped profile" err="rpc error: code = Internal desc = failed to ingest profile: normalize stacktraces: Txn is too big to fit into one request
```

Source https://github.com/dgraph-io/badger/blob/69926151f6532f2fe97a9b11ee9281519c8ec5e6/errors.go#L39

@brancz suggested to increase `maxRetries`, i set it to 10, this makes memory available, however the values seem off and I can't select a profile:

![image](https://user-images.githubusercontent.com/375856/190335909-1c20b230-7a09-4747-bd4a-82b542d15652.png)

and parca crashes shortly after it:
```
level=debug name=parca ts=2022-09-15T06:52:23.226803Z caller=scrape.go:320 scrape_pool=kcp msg="scraping profile" url=http://127.0.0.1:6060/debug/pprof/block
level=debug name=parca ts=2022-09-15T06:52:23.369931Z caller=scrape.go:468 scrape_pool=kcp target=http://127.0.0.1:6060/debug/pprof/block msg="appending new sample" labels="{instance=\"127.0.0.1:6060\", job=\"kcp\", __name__=\"block\"}"
level=debug name=parca ts=2022-09-15T06:52:23.370377Z caller=badger.go:450 msg=GetOrCreateStacktraces stacktrace_keys_len=0
level=debug name=parca ts=2022-09-15T06:52:23.370421Z caller=ingest.go:90 msg="no samples found in profile, dropping it" name=block sample_type=contentions sample_unit=count labels="{instance=\"127.0.0.1:6060\", job=\"kcp\"}"
level=debug name=parca ts=2022-09-15T06:52:23.370445Z caller=ingest.go:90 msg="no samples found in profile, dropping it" name=block sample_type=delay sample_unit=nanoseconds labels="{instance=\"127.0.0.1:6060\", job=\"kcp\"}"
level=debug name=parca ts=2022-09-15T06:52:28.49613Z caller=symbolizer.go:127 component=symbolizer msg="start symbolization cycle"
level=debug name=parca ts=2022-09-15T06:52:28.496488Z caller=symbolizer.go:153 component=symbolizer msg="no locations to symbolize"
level=debug name=parca ts=2022-09-15T06:52:28.496506Z caller=symbolizer.go:129 component=symbolizer msg="symbolization loop completed"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x103116de8]

goroutine 871 [running]:
github.com/polarsignals/frostdb.(*Granule).ColumnChunks(0x14010a863f0)
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/granule.go:378 +0x328
github.com/polarsignals/frostdb.(*ColumnRef).Column(0x140207ecd00, {0x106a92648, 0x14010a863f0})
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/binaryscalarexpr.go:108 +0x74
github.com/polarsignals/frostdb.BinaryScalarExpr.Eval({0x140207ecd00, 0x1, {0x14022b1f2aa, 0x6, 0xf9, 0x0, 0x0, 0x0}}, {0x106a92648?, 0x14010a863f0?})
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/binaryscalarexpr.go:130 +0x50
github.com/polarsignals/frostdb.(*AndExpr).Eval(0x1400b408a80, {0x106a92648, 0x14010a863f0})
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/filter.go:132 +0x38
github.com/polarsignals/frostdb.(*TableBlock).RowGroupIterator.func1({0x106a7e660?, 0x14010a863f0})
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/table.go:974 +0x6c
github.com/google/btree.(*node[...]).iterate(0x14005795840, 0x14011a11dd8, {{0x0?, 0x0?}, 0x48?}, {{0x0?, 0x0?}, 0xe0?}, 0x0?, 0x1, ...)
	/Users/surbania/go/pkg/mod/github.com/google/btree@v1.1.2/btree_generic.go:522 +0x308
github.com/google/btree.(*node[...]).iterate(0x14005795c00, 0x480, {{0x0?, 0x0?}, 0x20?}, {{0x0?, 0x0?}, 0xf0?}, 0x0?, 0x1, ...)
	/Users/surbania/go/pkg/mod/github.com/google/btree@v1.1.2/btree_generic.go:527 +0x394
github.com/google/btree.(*node[...]).iterate(0x14005795580, 0x14011a11f08, {{0x0?, 0x0?}, 0x38?}, {{0x0?, 0x0?}, 0x1?}, 0x0?, 0x0, ...)
	/Users/surbania/go/pkg/mod/github.com/google/btree@v1.1.2/btree_generic.go:527 +0x394
github.com/google/btree.(*BTreeG[...]).Ascend(0x0?, 0x0?)
	/Users/surbania/go/pkg/mod/github.com/google/btree@v1.1.2/btree_generic.go:779 +0x50
github.com/google/btree.(*BTree).Ascend(...)
	/Users/surbania/go/pkg/mod/github.com/google/btree@v1.1.2/btree_generic.go:1029
github.com/polarsignals/frostdb.(*TableBlock).RowGroupIterator(0x140002bc240?, {0x1400b0ce120?, 0x1400b0ce7b0?}, 0x104dd684d?, {0x106a7e5e0?, 0x1400b408a80?}, 0x14011a11fb8?)
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/table.go:970 +0x64
github.com/polarsignals/frostdb.(*Table).collectRowGroups(0x140002bc240, {0x106aa8ef0?, 0x1400b0ce7b0?}, 0x104dc5df6?, {0x106ab1a20, 0x1400b0ce120})
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/table.go:1293 +0x1fc
github.com/polarsignals/frostdb.(*Table).ArrowSchema(0x140002bc240, {0x106aa8ef0?, 0x1400b0ce780?}, 0x0?, {0x40?, 0x14011a12918?}, {{0x14000e65340, 0xb, 0xb}, {0x0, ...}, ...})
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/table.go:631 +0x3d4
github.com/polarsignals/frostdb/query/physicalplan.(*TableScan).Execute.func1({0x106aa8ef0, 0x1400b0ce780}, 0x1400b0ce750?)
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/query/physicalplan/physicalplan.go:116 +0x120
github.com/polarsignals/frostdb.(*Table).View(0x140002bc240, {0x106aa8ef0?, 0x1400b0ce750?}, 0x14010212a00)
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/table.go:441 +0x1c4
github.com/polarsignals/frostdb/query/physicalplan.(*TableScan).Execute(0x1400b0ce720, {0x106aa8ef0?, 0x1400b0ce210?}, {0x106a9a490?, 0x108801468})
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/query/physicalplan/physicalplan.go:115 +0x274
github.com/polarsignals/frostdb/query/physicalplan.(*OutputPlan).Execute(...)
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/query/physicalplan/physicalplan.go:83
github.com/polarsignals/frostdb/query.LocalQueryBuilder.Execute({{0x106a9a490, 0x108801468}, {0x106a88160, 0x108801468}, {0x140236becc0}}, {0x106aa8ef0?, 0x14000261f50?}, 0x140207ec710)
	/Users/surbania/go/pkg/mod/github.com/polarsignals/frostdb@v0.0.0-20220908165200-51ec92480a76/query/engine.go:141 +0x1ac
github.com/parca-dev/parca/pkg/parcacol.(*Querier).QueryRange(0x14000822d40, {0x106aa8ef0, 0x14000261f50}, {0x140023c2390?, 0x2d?}, {0x4?, 0x14000f212d8?, 0x0?}, {0x1?, 0x104da691c?, ...}, ...)
	/Users/surbania/src/parca/pkg/parcacol/querier.go:276 +0x70c
github.com/parca-dev/parca/pkg/query.(*ColumnQueryAPI).QueryRange(0x14000822d80, {0x106aa8ef0, 0x14000261f50}, 0x14010212690)
	/Users/surbania/src/parca/pkg/query/columnquery.go:97 +0x1b4
github.com/parca-dev/parca/gen/proto/go/parca/query/v1alpha1._QueryService_QueryRange_Handler.func1({0x106aa8ef0, 0x14000261f50}, {0x1068a1d00?, 0x14010212690})
	/Users/surbania/src/parca/gen/proto/go/parca/query/v1alpha1/query_vtproto.pb.go:197 +0x74
github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors.UnaryServerInterceptor.func1({0x106aa8ef0, 0x14000261f50}, {0x1068a1d00, 0x14010212690}, 0x0?, 0x1400ee35e78)
	/Users/surbania/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware/v2@v2.0.0-rc.2.0.20201207153454-9f6bf00c00a7/interceptors/server.go:22 +0x154
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x106aa8ef0?, 0x14000261f50?}, {0x1068a1d00?, 0x14010212690?})
	/Users/surbania/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3c
github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1({0x106aa8ef0, 0x14000261f50}, {0x1068a1d00, 0x14010212690}, 0x0?, 0x1400b408180)
	/Users/surbania/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:107 +0x74
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x106aa8ef0?, 0x14000261f50?}, {0x1068a1d00?, 0x14010212690?})
	/Users/surbania/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3c
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1({0x106aa8ef0, 0x14000261e60}, {0x1068a1d00, 0x14010212690}, 0x1400b408160, 0x1400b4081a0)
	/Users/surbania/go/pkg/mod/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.34.0/interceptor.go:325 +0x4dc
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x106aa8ef0?, 0x14000261e60?}, {0x1068a1d00?, 0x14010212690?})
	/Users/surbania/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3c
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x106aa8ef0, 0x14000261e60}, {0x1068a1d00, 0x14010212690}, 0x1400036aaa8?, 0x102e086e4?)
	/Users/surbania/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xb8
github.com/parca-dev/parca/gen/proto/go/parca/query/v1alpha1._QueryService_QueryRange_Handler({0x106856e40?, 0x14000822d80}, {0x106aa8ef0, 0x14000261e60}, 0x1400207f9d0, 0x140003aeb70)
	/Users/surbania/src/parca/gen/proto/go/parca/query/v1alpha1/query_vtproto.pb.go:199 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0x140010c61e0, {0x106abb6c0, 0x14008e62120}, 0x140010c59e0, 0x140003afe90, 0x10879fd80, 0x0)
	/Users/surbania/go/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:1301 +0x9c4
google.golang.org/grpc.(*Server).handleStream(0x140010c61e0, {0x106abb6c0, 0x14008e62120}, 0x140010c59e0, 0x0)
	/Users/surbania/go/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:1642 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/Users/surbania/go/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:938 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/surbania/go/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:936 +0x290
```

--- a/pkg/debuginfo/client.go
+++ b/pkg/debuginfo/client.go
@@ -34,8 +34,8 @@ const (
 	// uploaded and downloaded. AWS S3 has a minimum of 5MB for multi-part uploads
 	// and a maximum of 15MB, and a default of 8MB.
 	ChunkSize = 1024 * 1024 * 8
-	// MaxMsgSize is the maximum message size the server can receive or send. By default, it is 4MB.
-	MaxMsgSize = 1024 * 1024 * 32
+	// MaxMsgSize is the maximum message size the server can receive or send. By default, it is 64MB.
+	MaxMsgSize = 1024 * 1024 * 64
 )
 
 type Client struct {

--- a/pkg/metastore/badger.go
+++ b/pkg/metastore/badger.go
@@ -443,7 +443,7 @@ func (m *BadgerMetastore) GetOrCreateStacktraces(ctx context.Context, r *pb.GetO
 		stacktraceKeys = append(stacktraceKeys, MakeStacktraceKey(stacktrace))
 	}
 
-	const maxRetries = 2
+	const maxRetries = 100
 	var result retryableGetOrCreateStacktraces
 	var err error
 

